internal/reboot/reboot.go

```diff
@@ -2,25 +2,159 @@ package reboot
+	"github.com/openshift-kni/lifecycle-agent/internal/ostreeclient"
 	"github.com/openshift-kni/lifecycle-agent/lca-cli/ops"
 	rpmostreeclient "github.com/openshift-kni/lifecycle-agent/lca-cli/ostreeclient"
+	lcautils "github.com/openshift-kni/lifecycle-agent/utils"
+
+	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
 )
 
 var (
 	defaultRebootTimeout = 60 * time.Minute
 )
 
+type IBUAutoRollbackConfig struct {
+	InitMonitorEnabled bool              `json:"monitor_enabled,omitempty"`
+	InitMonitorTimeout int               `json:"monitor_timeout,omitempty"`
+	EnabledComponents  map[string]bool   `json:"enabled_components,omitempty"`
+	LcaTestVars        map[string]string `json:"lca_test_vars,omitempty"`
+}
+
+func writeInstallationConfigurationEnvFile(ostreeClient ostreeclient.IClient, stateroot, content string) error {
+	deploymentDir, err := ostreeClient.GetDeploymentDir(stateroot)
+	if err != nil {
+		return fmt.Errorf("unable to determine deployment dir: %w", err)
+	}
+
+	envFile := common.PathOutsideChroot(filepath.Join(deploymentDir, common.InstallationConfigurationEnvFile))
+	f, err := os.OpenFile(envFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		return fmt.Errorf("unable to open env file %s: %w", envFile, err)
+	}
+	defer f.Close()
+
+	if _, err := f.WriteString(content); err != nil {
+		return fmt.Errorf("unable to write to env file %s: %w", envFile, err)
+	}
+
+	return nil
+}
+
+func WriteIBUAutoRollbackConfigFile(ibu *lcav1alpha1.ImageBasedUpgrade, ostreeClient ostreeclient.IClient) error {
+	stateroot := common.GetStaterootName(ibu.Spec.SeedImageRef.Version)
+	staterootPath := common.GetStaterootPath(stateroot)
+	cfgfile := common.PathOutsideChroot(filepath.Join(staterootPath, common.IBUAutoRollbackConfigFile))
+
+	cfgdir := filepath.Dir(cfgfile)
+	if err := os.MkdirAll(cfgdir, 0o700); err != nil {
+		return fmt.Errorf("unable to create config dir: %s: %w", cfgdir, err)
+	}
+
+	monitorTimeout := ibu.Spec.AutoRollbackOnFailure.InitMonitorTimeoutSeconds
+	if monitorTimeout <= 0 {
+		monitorTimeout = common.IBUAutoRollbackInitMonitorTimeoutDefaultSeconds
+	}
+
+	rollbackCfg := IBUAutoRollbackConfig{
+		InitMonitorEnabled: !ibu.Spec.AutoRollbackOnFailure.DisabledInitMonitor,
+		InitMonitorTimeout: monitorTimeout,
+		EnabledComponents:  make(map[string]bool),
+		LcaTestVars:        make(map[string]string),
+	}
+
+	rollbackCfg.EnabledComponents["config"] = !ibu.Spec.AutoRollbackOnFailure.DisabledForPostRebootConfig
+	rollbackCfg.EnabledComponents["postpivot"] = !ibu.Spec.AutoRollbackOnFailure.DisabledForPostRebootConfig
+
+	// Check environ for LCA_TEST_* vars and add them
+	envFileContent := ""
+	re := regexp.MustCompile(`^LCA_TEST_`)
+	for _, envVar := range os.Environ() {
+		if re.MatchString(envVar) {
+			pair := strings.SplitN(envVar, "=", 2)
+			rollbackCfg.LcaTestVars[pair[0]] = pair[1]
+			envFileContent += envVar + "\n"
+		}
+	}
+
+	if envFileContent != "" {
```

Comment:
Condition seems unnecessary, let's write the file despite it being empty, for more consistent behavior